CMakeLists.txt

```diff
@@ -142,3 +145,50 @@ if(NOT CMAKE_SKIP_INSTALL_RULES)
     DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/assert
   )
 endif()
+
+# Don't run tests when library is used with add_subdirectory
+if(PROJECT_SOURCE_DIR STREQUAL CMAKE_SOURCE_DIR)
+  if(ASSERT_BUILD_TESTS)
+    include(CTest)
+
+    file(GLOB_RECURSE demo_sources tests/demo/*.cpp)
+    add_executable(demo ${demo_sources})
+    target_link_libraries(demo PRIVATE assert)
+    target_compile_options(
+      demo
+      PRIVATE
+      "-DASSERT_USE_MAGIC_ENUM"
+      "-DASSERT_FAIL=custom_fail"
+      "-DASSERT_LOWERCASE"
+    )
+
+    file(GLOB_RECURSE integration_sources tests/integration/*.cpp)
```

Comment:
You might want `CONFIGURE_DEPENDS` here, you're already using CMake 3.12 so it should be available.

Reply:
Thank you!

Reply:
I went ahead and just removed the glob entirely